### PR TITLE
fix multiple cursors on multiple screens

### PR DESCRIPTION
**Pull request by Dami-star**
When the cursor moves across the screen, the cursor on the first screen is not cleared, need to call set_cursor again to clear it.

### CODE REVIEW

File: src/server/qtquick/woutputrenderwindow.cpp [hunk 1]
```diff
@@ -1067,8 +1072,9 @@ bool OutputHelper::tryToHardwareCursor(const LayerData *layer)
                           ? layer->renderer->lastBuffer()->handle()
                           : nullptr;
         if (!buffer) {
-            if (set_cursor)
+            if (set_cursor && m_cursorRenderComplete)
```
Comment (zccrs) [inline src/server/qtquick/woutputrenderwindow.cpp:1075]: Don't care for `m_cursorRenderComplete` in this `if`.

File: src/server/qtquick/woutputrenderwindow.cpp [hunk 2]
```diff
@@ -826,6 +827,10 @@ WBufferRenderer *OutputHelper::afterRender()
 
     if (layers.isEmpty()) {
         cleanLayerCompositor();
+        if (m_cursorRenderComplete) {
+            tryToHardwareCursor(nullptr);
+            cleanCursorRender();
```
Comment (zccrs) [inline src/server/qtquick/woutputrenderwindow.cpp:832]: Call before `if (m_cursorRendercomplete)`

File: src/server/qtquick/woutputrenderwindow.cpp [hunk 3]
```diff
@@ -246,6 +246,7 @@ class Q_DECL_HIDDEN OutputHelper : public WOutputHelper
     QPointer<WBufferRenderer> m_cursorRenderer;
     BufferRendererProxy *m_cursorLayerProxy = nullptr;
     bool m_cursorDirty = false;
+    bool m_cursorRenderComplete = false;
```
Comment (zccrs) [inline src/server/qtquick/woutputrenderwindow.cpp:249]: `m_hardwareCursorRenderComplete`

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/qtquick/woutputrenderwindow.cpp (hunk 1 above)
Comment (Dami-star) [reply src/server/qtquick/woutputrenderwindow.cpp]: done

File: src/server/qtquick/woutputrenderwindow.cpp (hunk 2 above)
Comment (Dami-star) [reply src/server/qtquick/woutputrenderwindow.cpp]: done

File: src/server/qtquick/woutputrenderwindow.cpp (hunk 3 above)
Comment (Dami-star) [reply src/server/qtquick/woutputrenderwindow.cpp]: done

**Review by zccrs — APPROVED** [review]

**Pull request merged**